FirebaseAdmin/FirebaseAdmin/Auth/AbstractFirebaseAuth.cs

```diff
@@ -0,0 +1,587 @@
+// Copyright 2018, Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Google.Api.Gax;
+using Google.Apis.Util;
+
+namespace FirebaseAdmin.Auth
+{
+    /// <summary>
+    /// See <see cref="FirebaseAuth"/> and <see cref="TenantAwareFirebaseAuth"/>.
+    /// </summary>
+    public abstract class AbstractFirebaseAuth : IFirebaseService
```

Comment:
Implement `IFirebaseService` at `FirebaseAuth`. That is still the top-level service for this namespace.